frontend/readhistory.lua

```diff
@@ -198,6 +198,15 @@ function ReadHistory:getPreviousFile(current_file)
     end
 end
 
+function ReadHistory:getFileByDirectory(directory)
+    assert(self ~= nil)
+    for i=1, #self.hist do
+        if realpath(ffiutil.dirname(self.hist[i].file)) == realpath(directory) then
```

Comment:
Better compute `realpath(directory)` outside the loop, so it is called once instead of #hist times.